youtube_dl/extractor/gedidigital.py

```diff
@@ -0,0 +1,243 @@
+        if qualities:
+            qualities = qualities.group(2)
+            qualities = qualities.split(',') if qualities else ['.']
+            qualities = [i for i in qualities if i]
+        else:
+            qualities = [mobj.group(2)]
+
+        http_formats = []
+        for q in qualities:
+            w, h, t = get_format_info(q)
+            http_formats.append({
+                'url': mp4['mp4'].replace(pattern, 'rrtv-%s-' % q),
+                'width': w or _QUALITY[q][0],
+                'height': h or _QUALITY[q][1],
+                'tbr': t or int(q),
+                'protocol': 'https',
+                'format_id': 'https-%s' % q,
+            })
+        return http_formats
+
+    def _real_extract(self, url):
+        u = re.match(self._VALID_URL, url)
+        self.IE_NAME = u.group('iename') if u.group('iename') else 'gedi'
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_id)
+        player_data = re.findall(
+            r'PlayerFactory\.setParam\(\'(?P<type>.+?)\',\s*\'(?P<name>.+?)\',\s*\'(?P<val>.+?)\'\);',
+            webpage)
+
+        formats = []
+        audio_fmts = []
+        title = None
+        thumb = None
+        mp4 = {}
+
+        for t, n, v in player_data:
+            if t == 'format':
+                if n == 'mp4':
+                    mp4.update({'mp4': v})
+                if n == 'video-hls-vod-ak':
+                    mp4.update({'manifest': v})
+                    formats.extend(self._extract_akamai_formats(
+                        v, video_id, {'http': 'media.gedidigital.it'}))
+                if n == 'audio-hls-vod':
+                    audio_fmts.extend(self._extract_m3u8_formats(
+                        v, video_id, 'm4a', m3u8_id='audio-hls', fatal=False))
+                if n == 'audio-rrtv-mp3':
+                    audio_fmts.append({
+                        'format_id': 'mp3',
+                        'url': v,
+                        'tbr': 128,
+                        'ext': 'mp3',
+                        'vcodec': 'none',
+                        'acodec': 'mp3',
+                    })
+            elif t == 'param':
+                if n == 'videotitle':
+                    title = v
+                if n in ['image_full_play', 'image_full', 'image']:
+                    thumb = v
+
+        title = self._og_search_title(webpage) if not title else title
+
+        # clean weird char
+        title = compat_str(title).encode('utf8', 'replace').replace(b'\xc3\x82', b'').decode('utf8', 'replace')
+
+        self._clean_audio_fmts(audio_fmts)
+        formats.extend(audio_fmts)
+
+        if mp4:
+            formats.extend(self._generate_http_urls(mp4, formats) or [])
+
+        self._sort_formats(formats)
+
+        return {
+            'id': video_id,
+            'title': title,
+            'description': self._html_search_meta('twitter:description', webpage),
```

Comment:
add more fallbacks and extract `timestamp`.

Reply:
done